lib/validate.rb

```diff
@@ -0,0 +1,58 @@
+require './output'
+
+module Validate
+  include Output
+  def is_valid?(the_input)
+    valid = true
+    errors = []
+
+    if !length_valid(the_input)
+      valid = false
+      errors << 'length'
+    end
+
+    if !type_valid(the_input)
+      valid = false
+      errors << 'type'
+    end
+
+    if !context_valid(the_input)
+      valid = false
+      errors << 'context'
+    end
+
+    if !valid
+      print_error_start
+
+      errors.each do |error|
+        if error.eql? 'length'
+          print_length_error
+        elsif error.eql? 'type'
+          print_type_error
+        elsif error.eql? 'context'
+          print_context_error
+        end
```

Comment:
Let's break off the printing of each error into a seperate method. I like the chaining you're doing here but I think you can separate the handling of recording errors and printing them out to the console.